src/wasi.rs

```diff
@@ -8,12 +8,18 @@
 
 //! Implementation for WASI
 use crate::Error;
-use core::{mem::MaybeUninit, num::NonZeroU32};
-use wasi::wasi_snapshot_preview1::random_get;
+use core::{
+    mem::MaybeUninit,
+    num::{NonZeroU16, NonZeroU32},
+};
+use wasi::random_get;
 
 pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
-    match unsafe { random_get(dest.as_mut_ptr() as i32, dest.len() as i32) } {
-        0 => Ok(()),
-        err => Err(unsafe { NonZeroU32::new_unchecked(err as u32) }.into()),
-    }
+    unsafe { random_get(dest.as_mut_ptr() as *mut u8, dest.len()) }.map_err(|e| {
+        // The WASI errno will always be non-zero, but we check just in case.
+        match NonZeroU16::new(e.raw()) {
+            Some(r) => Error::from(NonZeroU32::from(r)),
+            None => Error::ERRNO_NOT_POSITIVE,
```

Comment:
One minor problem with this code is that compiler will not be able to remove the `None` branch without LTO. [`random_get`](https://github.com/bytecodealliance/wasi/blob/main/src/lib_generated.rs#L2056) is not marked as `#[inline]` so compiler only knows that `wasi::Errno` has type `u16`. But I guess it's fine, since the error should not happen in the first place.

Reply:
This is correct, but I think it's fine because failure is rare, and the branch will be extremely predictable.